lib/states/destroy_card.rb

```diff
@@ -0,0 +1,37 @@
+module States
+  class DestroyCard < BaseState
+    MENU_STATE = :main_menu_message
+
+    def step
+      return MenuAccount.new(@context) if @next_state == MENU_STATE
+
+      DestroyCard.new(@context)
+    end
+
+    def action
+      return @next_state = MENU_STATE unless account_have_cards?(@context.extant_account.card)
```

Comment:
Can we move account_have_cards?(@context.extant_account.card) to private method& and use it in step and action?

Reply:
instead @next_state